apt_offline_core/AptOfflineCoreLib.py

```diff
@@ -2200,6 +2227,9 @@ def main():
         parser_get.add_argument("--https-key-file", dest="https_key_file",
 						help="Certificate key for https client authentication", type=str, default=None)
 
+        parser_get.add_argument("--http-basicauth", dest="http_basicauth",  action='append',
+						help="A user/password for a certain domain in format: 'user:password@domain'. This argument can be passed as many times as required", type=str, default=[])
```

Comment:
I would rather prefer this come as a full URL like: https://username:password%23%40%21@apt-offline.githut.com:80/
Then, parse and sanitize it through `urlparse`

Something like:

```
In [42]: url = "https://username:password%23%40%21@apt-offline.githut.com:80/"

In [43]: parsed_url = urlparse(url)

In [44]: parsed_url.scheme
Out[44]: 'https'

In [45]: parsed_url.username
Out[45]: 'username'

In [46]: parsed_url.password
Out[46]: 'password%23%40%21'

In [47]: urllib.parse.unquote(parsed_url.password)
Out[47]: 'password#@!'

In [48]: parsed_url.geturl()
Out[48]: 'https://username:password%23%40%21@apt-offline.githut.com:80/'

In [49]: parsed_url.port
Out[49]: 80

```

This will take care of many validations and also for special percent encoded password characters. Refer **RFC-3986 section 2.1**